doc/Data Plane Code Generation/DeveloperDrivenEvolution.md

```diff
@@ -0,0 +1,145 @@
+# Developer driven evolution
+To add a convenience API that returns raw JSON to a model, steps are:
+
+**Get convenience method signiture from protocol method signiture.** Replace the optional `RequestContext` parameter with an optional `CancellationToken`. If there is an input `RequestContent`, replace it with a model type `T`. Replace the return value `Response` with `Response<T>`.
+```C#
+- public virtual async Task<Response> CreateMetricFeedbackAsync(RequestContent content, RequestContext context = null);
++ public virtual async Task<Response<MetricFeedback>> CreateMetricFeedbackAsync(MetricFeedback feedback, CancellationToken cancellationToken = default);
+```
+**Pick the best convenience method name fitting your scenario.** You could use the same method name as the initial protocol method only when your convenience method has different **required** parameter list. E.g.,
```

Comment:
I think partner service teams following this guidance will think we are telling them to "pick a new name for the method."  

Let's include this as a sub-bullet of "Get convenience method signature," where the steps are:
1. Replace input parameters
2. Replace output value
3. Rename method only if needed
    - Only needed if ambiguous call site
    - Suggested rename for most cases
    - If this name doesn't make sense, pick the best name

It would be better in the explanation to show full method signatures rather than parameter lists - it's unclear what the parameter lists are trying to describe.